### PR TITLE
use 256 colour codes in @themerdev/vim-lightline

### LINKED ISSUE #118: vim-lightline: colors not used in CLI
# Description
The vim-lightline theme affects vim GUIs that use it, but not CLI like ... vim.

# To reproduce

```
yarn add themer @themerdev/vim-lightline @themerdev/colors-default
yarn run themer -c @themerdev/colors-default -t @themerdev/vim-lightline -o build
mkdir -p ~/.vim/autoload/lightline/colorscheme/
cp build/vim-lightline/ThemerVimLightline.vim ~/.vim/autoload/lightline/colorscheme/
echo "let g:lightline = { 'colorscheme': 'ThemerVimLightline' }" >> ~/.vimrc
```

## Screenshots
That's how it should be :
NORMAL is light blue
That's how it is (check the NORMAL background being darker) :
NORMAL should be light blue

## Expected behavior
I expect lightline's theme to have support for both CLI and GUI applications.

# Environment information

## CLI

Running up-to-date Arch linux.
node version 16.11.1
themer version ^3.3.9
@themerdev/vim-lightline version ^3.1.2

### PR DESCRIPTION
**Pull request by lafleurdeboum**
Resolves  #118 .

I propose to use 256 colour-codes in vim-lightline, because CLIs like `vim` and `nvim` don't support html colour notation. This may lead to huge discrepancies, highlighting codes not being read at all.

I understand using this lower definition may lead to subtle discrepancies between terminal and vim GUIs, but I consider it a lower price to pay. Hope some would agree ?

I copied the decoding algorithm from @themerdev/vim.

### CODE REVIEW

**Comment by mjswensen** [comment]
Thank you, @lafleurdeboum , for your effort on this! I like the idea of being able to support console Vim in addition to users setting `termguicolors`.

How about both? Since the `render()` function in themer packages returns an array of files, we could ship `ThemerVimLightline.vim` with the current color notation (to keep the package backwards compatible) _and_ `ThemerVimLightline.cterm.vim` with the 256 color codes as you have it here. What do you think?

**Comment by lafleurdeboum** [comment]
Actually, running
```
:let g:lightline#colorscheme#default#palette
```
shows that lightline colorscheme supports both code systems in the same
dictionary (I don't know how vimscript calls its dictionaries, sorry).
So `render()` could return both. I've tested this feature and updated
the MR accordingly.

 
Le vendredi 12 novembre 2021 à 06:01 -0800, Matt Swensen a écrit :

File: cli/packages/vim-lightline/lib/index.js [hunk 1]
```diff
@@ -41,21 +85,21 @@ const render = (colors) => {
...
-  let s:p.tabline.left = [ [ s:shade6, s:shade2 ] ]
-  let s:p.tabline.tabsel = [ [ s:shade6, s:shade0 ] ]
-  let s:p.tabline.middle = [ [ s:shade2, s:shade4 ] ]
+  let s:p.normal.left = [ [ s:guishade1, s:guiaccent5, s:shade1, s:accent5 ], [ s:guishade7, s:guishade2, s:shade7, s:shade2 ] ]
+  let s:p.normal.right = [ [ s:guishade1, s:guishade4, s:shade1, s:shade4 ], [ s:shade5, s:shade2 ] ]
+  let s:p.inactive.right = [ [ s:guishade1, s:guishade3, s:shade1, s:shade3 ], [ s:shade3, s:shade1 ] ]
+  let s:p.inactive.left =  [ [ s:guishade4, s:guishade1, s:shade4, s:shade1 ], [ s:shade3, s:shade0 ] ]
+  let s:p.insert.left = [ [ s:guishade1, s:guiaccent3, s:shade1, s:accent3 ], [ s:shade7, s:shade2 ] ]
+  let s:p.replace.left = [ [ s:guishade1, s:guiaccent1, s:shade1, s:accent1 ], [ s:shade7, s:shade2 ] ]
+  let s:p.visual.left = [ [ s:guishade1, s:guiaccent6, s:shade1, s:accent6 ], [ s:shade7, s:shade2 ] ]
```
Comment (mjswensen) [inline cli/packages/vim-lightline/lib/index.js:94]: Do we need to add the `gui` ones to the second list for these lines as well (as you have done on line 88), or no?

**Comment by mjswensen** [comment]
> Actually, running ``` :let g:lightline#colorscheme#default#palette ``` shows that lightline colorscheme supports both code systems in the same dictionary (I don't know how vimscript calls its dictionaries, sorry). So `render()` could return both. I've tested this feature and updated the MR accordingly.

Fantastic!! Thank you!

File: cli/packages/vim-lightline/lib/index.js (hunk 1 above)
Comment (lafleurdeboum) [reply cli/packages/vim-lightline/lib/index.js:94]: Well, `:let g:lightline#colorscheme#default#palette` has them, so I guess we do need them to comply with lightline.

**Comment by lafleurdeboum** [comment]
I pushed a modified last commit with a correction from `prettier` ; the CI should succeed now.

**Comment by mjswensen** [comment]
Sorry for the delay on this, @lafleurdeboum — these changes have now been published at [`@themerdev/vim-lightline`](https://www.npmjs.com/package/@themerdev/vim-lightline) version `3.2.0`. ✅

**Pull request merged**